conans/test/util/tools_test.py

```diff
@@ -576,6 +576,33 @@ def vcvars_constrained_test(self):
                 # Not raising
                 tools.vcvars_command(settings, force=True)
 
+    def vcvars_context_manager_test(self):
+        conanfile = """
+from conans import ConanFile, tools
+
+class MyConan(ConanFile):
+    name = "MyConan"
+    version = "0.1"
+    settings = "os", "compiler"
+
+    def build(self):
+        with tools.vcvars(self.settings):
+            self.output.info(str(self.settings.os) + ": vcvars applies")
+            self.output.info("VCINSTALLDIR set to: " + str(tools.get_env("VCINSTALLDIR")))
+"""
+        client = TestClient()
+        client.save({"conanfile.py": conanfile})
+
+        if platform.system() == "Windows":
+            client.run("create . conan/testing")
+            self.assertIn("MyConan/0.1@conan/testing: Windows: vcvars applies", client.out)
+            self.assertNotIn("VCINSTALLDIR set to: None", client.out)
+
+        if platform.system() == "Linux":
```

Comment:
Let OSX run something. instead of `if platform.system() == "Linux"` run an else. Probably the OS name printed is pointless.